server/src/internalClusterTest/java/org/elasticsearch/reservedstate/service/FileSettingsServiceIT.java

```diff
@@ -498,6 +502,63 @@ public void testSettingsAppliedOnMasterReElection() throws Exception {
         assertClusterStateSaveOK(savedClusterState.v1(), savedClusterState.v2(), "43mb");
     }
 
+    public void testHealthIndicator() throws Exception {
+        internalCluster().setBootstrapMasterNodeIndex(0);
+        logger.info("--> start a data node to act as the health node");
+        String healthNode = internalCluster().startNode(
+            Settings.builder().put(dataOnlyNode()).put("discovery.initial_state_timeout", "1s")
+        );
+
+        logger.info("--> start master node");
+        final String masterNode = internalCluster().startMasterOnlyNode(
+            Settings.builder().put(INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0s").build()
+        );
```

Comment:
On second thought, I realize the `dataOnlyNode` and `masterOnlynode` do actually enforce which node will become the health node (because we filter by `DiscoveryNode::canContainData` when assigning the health node). Apologies for the confusion... I think having some randomization of the health node is something we want, as it allows us to cover both scenarios of the master being the health node and another node being the health node. What do you think of just starting two nodes and being completely dependent on which node has which role? (we can still obtain the current master in the cluster of course, so nothing really changes in this test other than that)

Reply:
If there are two cases that matter, I'd be inclined just to test both. That way, when the tests pass, you have confidence the system works.

(But then I've never really grasped our exact philosophy behind randomized testing. I thought it was a way to test an impractically large number of cases, but in ES we seem to use it as our default way to get test coverage.)

I'm happy to do whatever the "usual" thing is here. Consistency with the rest of the codebase is a big plus.

Reply:
I'm also fine splitting it up into two tests. I agree that the use of randomization is a bit vague here sometimes. It's supposed to be used as a "I don't care what this is" but we indeed have a tendency to use it to increase coverage.